### PR TITLE
Add trento architecture guide and assets

### PR DESCRIPTION
**Pull request by arbulu89**
Update the trento-architecture diagram and bring the content from the web repository.
Once this is merged, the content in the web repository will be deleted.

The `md` file content is a copy/paste except the `Checks Execution` chapter

Have a look at: https://github.com/trento-project/docs/blob/trento-architecture/guides/architecture/trento-architecture.md

### CODE REVIEW

**Review by nelsonkopliku — COMMENTED** [review]
Great Job!
I would like to redeem myself from the mistakes of the past, so I left a comment :see_no_evil:

File: guides/architecture/trento-architecture.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+The Checks Execution is managed by the [Checks Engine](https://github.com/trento-project/wanda). It includes the Trento configuration checks catalog, which are written as `yaml` files. Each check includes a script written in [Rhai](https://rhai.rs/) which is what the check tries to evaluate with some facts coming from the targets.
+
+This component is responsible of:
+- Listening for checks execution requests coming from the control plane (or any other external user)
+- Sending the fact gathering requests to the appropriate targets
+- Evaluating check results using the gathered facts
+- Publishing the evaluated results
+- Storing and exposing the executed executions
+
+## Not only discovery
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply guides/architecture/trento-architecture.md:60]: Fair enough!

[the remark this answers is not in the document]
Comment (fabriziosestito) [reply guides/architecture/trento-architecture.md:60]: just my 2 cents, not sure hexagonal architecture is very relevant as it is implemented already by Phoenix, and intrinsically by using CQRS the business logic is unaware on how the  data is stored.
if you want to mention cqrs, I would rather link to commanded repository and skip the martin fowler link,

[the remark this answers is not in the document]
Comment (nelsonkopliku) [reply guides/architecture/trento-architecture.md:60]: Yes, a [link to commanded](https://github.com/commanded/commanded) is a great idea! :+1:

**Review by nelsonkopliku — APPROVED** [review]
Super!

File: guides/architecture/trento-architecture.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (arbulu89) [reply guides/architecture/trento-architecture.md]: Last chapters update. I have removed many parts and only leave some relevant information.

**Review by EMaksy — APPROVED** [review]
Well done, it is well written and understandable!

**Pull request merged**